Sources/NostrSDK/EventCreating.swift

```diff
@@ -291,6 +291,67 @@ public extension EventCreating {
         return try MuteListEvent(content: encryptedContent ?? "", tags: publicTags, signedBy: keypair)
     }
     
+    /// Creates a ``BookmarksListEvent`` (kind 10003) containing an uncategorized, "global" list of things a user wants to save.
+    /// - Parameters:
+    ///   - publiclyBookmarkedEventIds: Event ids to bookmark.
+    ///   - privatelyBookmarkedEventIds: Event ids to privately bookmark.
+    ///   - publiclyBookmarkedArticlesCoordinates: Articles coordinates to bookmark.
+    ///   - privatelyBookmarkedArticlesCoordinates: Articles coordinates to privately bookmark.
+    ///   - publiclyBookmarkedHashtags: Hashtags to bookmark.
+    ///   - privatelyBookmarkedHashtags: Hashtags to privately bookmark.
+    ///   - publiclyBookmarkedLinks: Links to bookmark.
+    ///   - privatelyBookmarkedLinks: Links to privately bookmark.
+    func bookmarksList(withPubliclyBookmarksEventIds publiclyBookmarkedEventIds: [String] = [],
+                       privatelyBookmarkedEventIds: [String] = [],
+                       publiclyBookmarkedArticlesCoordinates: [EventCoordinates] = [],
+                       privatelyBookmarkedArticlesCoordinates: [EventCoordinates] = [],
+                       publiclyBookmarkedHashtags: [String] = [],
+                       privatelyBookmarkedHashtags: [String] = [],
+                       publiclyBookmarkedLinks: [URL] = [],
+                       privatelyBookmarkedLinks: [URL] = [],
+                       signedBy keypair: Keypair) throws -> BookmarksListEvent {
+        var publicTags = [Tag]()
+        
+        for eventId in publiclyBookmarkedEventIds {
+            publicTags.append(.event(eventId))
+        }
+        for coordinates in publiclyBookmarkedArticlesCoordinates {
+            publicTags.append(coordinates.tag)
+        }
+        for hashtag in publiclyBookmarkedHashtags {
+            publicTags.append(.hashtag(hashtag))
+        }
+        for link in publiclyBookmarkedLinks {
+            publicTags.append(Tag(name: .webURL, value: link.absoluteString))
+        }
+        
+        var secretTags = [[String]]()
```

Comment:
It makes me a bit nervous to handle raw strings (as opposed to how we're handling the public tags) because we lose out on the existing structure and validation.

Reply:
We have to build them into the "raw" string structure here to prepare for encryption, but maybe we could have each Tag render its own "raw" tag, which would be the array of strings. I'll work on that a bit.